pkg/protecode/protecode.go

```diff
@@ -371,23 +382,35 @@ func (pc *Protecode) UploadScanFile(cleanupMode, group, filePath, fileName, vers
 }
 
 // DeclareFetchURL configures the fetch url for the protecode scan
-func (pc *Protecode) DeclareFetchURL(cleanupMode, group, fetchURL, version string, productID int, replaceBinary bool) *ResultData {
+func (pc *Protecode) DeclareFetchURL(cleanupMode, group, customDataJSONMap, fetchURL, version string, productID int, replaceBinary bool) *ResultData {
 	deleteBinary := (cleanupMode == "binary" || cleanupMode == "complete")
 
 	var headers = make(map[string][]string)
+	if len(customDataJSONMap) > 0 {
+		customDataHeaders := map[string]string{}
+		if err := json.Unmarshal([]byte(customDataJSONMap), &customDataHeaders); err != nil {
+			log.Entry().Warn("[WARN] ===> customDataJSONMap flag must be a valid JSON map. Check the value of --customDataJSONMap and try again.")
+		} else {
+			for k, v := range customDataHeaders {
+				headers["META-"+strings.ToUpper(k)] = []string{v}
+			}
+		}
+	}
 
+	headers["Group"] = []string{group}
+	headers["Delete-Binary"] = []string{fmt.Sprintf("%v", deleteBinary)}
+	headers["Url"] = []string{fetchURL}
+	headers["Content-Type"] = []string{"application/json"}
 	if (replaceBinary) && (version != "") {
 		log.Entry().Debugf("[DEBUG][FETCH_URL] ===> replaceBinary && version != empty ")
-		headers = map[string][]string{"Group": {group}, "Delete-Binary": {fmt.Sprintf("%v", deleteBinary)}, "Replace": {fmt.Sprintf("%v", productID)}, "Version": {version}, "Url": {fetchURL}, "Content-Type": {"application/json"}}
+		headers["Replace"] = []string{fmt.Sprintf("%v", productID)}
+		headers["Version"] = []string{version}
 	} else if replaceBinary {
 		log.Entry().Debugf("[DEBUG][FETCH_URL] ===> replaceBinary")
-		headers = map[string][]string{"Group": {group}, "Delete-Binary": {fmt.Sprintf("%v", deleteBinary)}, "Replace": {fmt.Sprintf("%v", productID)}, "Url": {fetchURL}, "Content-Type": {"application/json"}}
+		headers["Replace"] = []string{fmt.Sprintf("%v", productID)}
 	} else if version != "" {
 		log.Entry().Debugf("[DEBUG][FETCH_URL] ===> version != empty ")
-		headers = map[string][]string{"Group": {group}, "Delete-Binary": {fmt.Sprintf("%v", deleteBinary)}, "Version": {version}, "Url": {fetchURL}, "Content-Type": {"application/json"}}
```

Comment:
`headers["Version"]` is missing here as well, please look at `else` condition

Reply:
Sorry about that -- I should have left a comment here to explain these changes.

I think these changes are functionally equivalent to the current implementation, however it removes the need to repeat yourself when initializing the map.

For example in the current source, we are sort of repeating ourselves in all conditions with 
```
 map[string][]string{"Group": {group}, "Delete-Binary": {fmt.Sprintf("%v", deleteBinary)}
```
etc...

Instead of copying the same code in each condition, I thought it would be more readable to initialize the headers that are present in all conditions first, and then add the condition-specific headers one by one.

With this approach, the `else` block isn't needed because the default headers were set at L400-403 instead.

```
	headers["Group"] = []string{group}
	headers["Delete-Binary"] = []string{fmt.Sprintf("%v", deleteBinary)}
	headers["Url"] = []string{fetchURL}
	headers["Content-Type"] = []string{"application/json"}
	if (replaceBinary) && (version != "") {
		log.Entry().Debugf("[DEBUG][FETCH_URL] ===> replaceBinary && version != empty ")
		headers["Replace"] = []string{fmt.Sprintf("%v", productID)}
		headers["Version"] = []string{version}
	} else if replaceBinary {
		log.Entry().Debugf("[DEBUG][FETCH_URL] ===> replaceBinary")
		headers["Replace"] = []string{fmt.Sprintf("%v", productID)}
	} else if version != "" {
		log.Entry().Debugf("[DEBUG][FETCH_URL] ===> version != empty ")
		headers["Version"] = []string{version}
	}
```